frontend/src/utils/YupSchema.tsx

```diff
@@ -50,7 +50,7 @@ export const Address = Yup.object().shape({
     .nullable(),
   provinceId: Yup.string().required('Required'),
   postal: Yup.string().matches(
-    /^[a-zA-z][0-9][a-zA-z]\s*?[0-9][a-zA-z][0-9]$/,
+    /^[a-zA-z][0-9][a-zA-z]\s*-*[0-9][a-zA-z][0-9]$/,
```

Comment:
Had to add the hyphen and space to the schema because it would run the validation before the input was formatted resulting in invalid validation. So even though the hyphen and space will not be present when the user clicks away the schema still saw it. Please let me know if there is a better way!